### DISCUSSION TITLE
Span names like `io.opentelemetry.contrib.mongodb`

### DISCUSSION THREAD

**Discussion by flaviut** [open-telemetry/opentelemetry-java-instrumentation#9328] [Q&A]
The specification for tracers suggests that the name uniquely identify the instrumentation library, [for example, with names like `io.opentelemetry.contrib.mongodb`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#get-a-tracer).

Using `-javaagent:opentelemetry-javaagent.jar` with Spring Boot and viewing the data in SigNoz/Clickhouse, I'm however seeing names like `SELECT postgres.permission`.

Is there some way to instead see a name more like `io.opentelemetry.contrib.postgres` or `db.postgres.select`? Ditto for HTTP requests: I can always look at the `db.statement` or `http.route` for details on what got called, but there's no way for me to query "all the HTTP calls".

**Comment by laurit** [comment]
The link you provided talks about the name of the `tracer`(https://opentelemetry.io/docs/specs/otel/trace/api/#tracer) rules governing the span name are different for database(https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md) and http server(https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#name) If you wish to alter these names you could either use a custom exporter that modifies the span name before passing it to the otlp exporter or see whether collector can perform the transformation you desire.

**Reply by flaviut** [reply]
I see, that clears that up! Thank you.
